packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart

```diff
@@ -379,22 +379,24 @@ gmaps.MarkerOptions _markerOptionsFromMarker(
   Marker marker,
   gmaps.Marker currentMarker,
 ) {
-  final iconConfig = marker.icon.toJson() as List;
+  final iconConfig = marker.icon?.toJson() as List;
   gmaps.Icon icon;
 
-  if (iconConfig[0] == 'fromAssetImage') {
-    // iconConfig[2] contains the DPIs of the screen, but that information is
-    // already encoded in the iconConfig[1]
+  if (iconConfig != null) {
+    if (iconConfig[0] == 'fromAssetImage') {
```

Comment:
nits:
Add `assert(iconConfig.length >= 2)`

Reply:
I have to add this inside of the iconConfig[0] == 'fromAssetImage', other iconConfigs might not be as long (like the default icon)